Sources/StreamChatUI/ChatMessageList/RETHINK/MessageCell.swift

```diff
@@ -5,331 +5,61 @@
-                reactionsView.topAnchor.pin(equalTo: contentView.topAnchor),
-                reactionsView.centerXAnchor.pin(
-                    equalTo: options.contains(.flipped) ? bubbleView.leadingAnchor : bubbleView.trailingAnchor
-                ),
-                bubbleView.topAnchor.pin(equalTo: reactionsView.centerYAnchor)
-            ]
-        }
-        
-        NSLayoutConstraint.activate(constraintsToActivate)
+    var content: _ChatMessage<ExtraData>? {
+        get { messageContentView.content }
+        set { messageContentView.content = newValue }
     }
 
-    override func updateContent() {
-        // Text
-        textView?.text = content?.text
-        
-        // Avatar
-        let placeholder = uiConfig.images.userAvatarPlaceholder1
-        if let imageURL = content?.author.imageURL {
-            authorAvatarView?.imageView.loadImage(from: imageURL, placeholder: placeholder)
-        } else {
-            authorAvatarView?.imageView.image = placeholder
-        }
-        
-        // Bubble view
-        if content?.type == .ephemeral {
-            bubbleView?.backgroundColor = uiConfig.colorPalette.popoverBackground
-            
-        } else if layoutOptions?.contains(.linkPreview) == true {
-            bubbleView?.backgroundColor = uiConfig.colorPalette.highlightedAccentBackground1
-            
-        } else {
-            bubbleView?.backgroundColor = content?.isSentByCurrentUser == true ?
-                uiConfig.colorPalette.background2 :
-                uiConfig.colorPalette.popoverBackground
-        }
-
-        let defaultAttachments = content?.attachments.compactMap { $0 as? ChatMessageDefaultAttachment }
-
-        // Metadata
-        metadataView?.content = content
-
-        // Link preview
-        linkPreviewView?.content = defaultAttachments?.first { $0.type.isLink }
-
-        // Quoted message view
-        quotedMessageView?.message = content?.quotedMessage
+    var delegate: MessageContentView<ExtraData>.Delegate? {
+        get { messageContentView.delegate }
+        set { messageContentView.delegate = newValue }
+    }
 
-        let attachments: _ChatMessageAttachmentListViewData<ExtraData>? = defaultAttachments.map {
-            .init(
-                attachments: $0,
-                didTapOnAttachment: { [weak self] in
-                    self?.delegate?.didTapOnAttachment($0)
-                },
-                didTapOnAttachmentAction: { [weak self] in
-                    self?.delegate?.didTapOnAttachmentAction($0, $1)
-                }
-            )
+    func setUpLayoutIfNeeded(options: ChatMessageLayoutOptions) {
+        if messageContentView.layoutOptions == nil {
+            let alignment = options.contains(.flipped) ?
+                messageContentView.trailingAnchor.pin(equalTo: contentView.trailingAnchor) :
+                messageContentView.leadingAnchor.pin(equalTo: contentView.leadingAnchor)
+            alignment.isActive = true
         }
```

Comment:
To me it feels the container should be responsible for content alignment, not the content view. That's why we set the leading/trailing constraint in a cell. However, the `if options == nil` check doesn't look nice but didn't find a better way to set the constraint just once